cruise-control-metrics-reporter/src/test/java/com/linkedin/kafka/cruisecontrol/metricsreporter/utils/CCEmbeddedKraftCluster.java

```diff
@@ -0,0 +1,297 @@
+   *
+   * @return a {@link Network} instance corresponding to either the existing Docker network (from env var)
+   * or a newly created network.
+   */
+  private Network getNetwork() {
+    String envNetworkName = System.getenv(NETWORK_NAME);
+    if (envNetworkName != null) {
+      return Network.builder().id(envNetworkName).build();
+    } else {
+      return Network.newNetwork();
+    }
+  }
+
+  /**
+   * Returns the Docker host address to be used for connecting to containers from the test environment.
+   *
+   * <p>This method first checks for the presence of a Docker host address in the environment variable
+   * specified by {@code CONTAINER_HOST_ENV_VAR}. This environment variable should be set externally,
+   * for example, in a CI environment like CircleCI, by finding the Docker network gateway IP
+   *
+   * @return the Docker host IP address or hostname to connect to containers.
+   */
+  private String getContainerHost() {
+    String envNetworkName = System.getenv(CONTAINER_HOST_ENV_VAR);
+    if (envNetworkName != null) {
+      return envNetworkName;
+    } else {
+      return DEFAULT_CONTAINER_HOST;
+    }
+  }
+
+  private void copyCertToContainer(KafkaContainer container, Map<Object, Object> config, String key) {
+    if (config.containsKey(key)) {
+      String path = config.get(key).toString();
+      container.withCopyToContainer(MountableFile.forHostPath(path, 0644), path);
+    }
+  }
+
+  private void setupContainerResources(KafkaContainer kafkaContainer, Map<Object, Object> brokerConfig) {
+    Path libsDir = Paths.get("build", "libs").toAbsolutePath().normalize();
+
+    try {
+      Path jarPath = Files.list(libsDir)
+        .filter(path -> path.getFileName().toString().startsWith("cruise-control-metrics-reporter"))
+        .filter(path -> path.getFileName().toString().endsWith(".jar"))
+        .findFirst()
+        .orElseThrow(() -> new IllegalStateException("Cruise Control Metrics Reporter jar not found in: " + libsDir));
+
+      kafkaContainer.withCopyToContainer(
+        MountableFile.forHostPath(jarPath.toString(), 0755),
+        "/opt/kafka/libs/cruise-control-metrics-reporter.jar"
+      );
+
+      copyCertToContainer(kafkaContainer, brokerConfig, "ssl.truststore.location");
+      copyCertToContainer(kafkaContainer, brokerConfig, "ssl.keystore.location");
+      copyCertToContainer(kafkaContainer, brokerConfig, "cruise.control.metrics.reporter.ssl.keystore.location");
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to mount Kafka container resources", e);
+    }
+  }
+
+  /**
+   * Updates the broker configuration of the Kafka container.
+   *
+   * @param kafkaContainer The Kafka container to be updated.
+   * @param brokerConfig A map of properties that will override the default broker
+   *                        configuration.
+   */
+  public static void setBrokerConfigViaEnvVars(KafkaContainer kafkaContainer, Map<Object, Object> brokerConfig) {
```

Comment:
This an odd API.

* 'ViaEnvVars' in the name is leaking information about the internal implementation.
* 'setBrokerConfig" lies - it is overriding the configuration, not setting it (with implies replacement).
* static is smelly,  maybe `overrideBrokerConfig(int nodeId, Map<Object, Object> brokerConfig)

Reply:
Good call, just fixed.